### PR TITLE
fix dia chain extension

### PR DESCRIPTION
**Pull request by ashneverdawn**
The chain extension inputs need to be fixed length, but the call to get_coin_info() expects variable length vecs. With this PR, the trailing zeros in the inputs are trimmed, so that get_coin_info() receives the expected vecs with the correct lengths.

### CODE REVIEW

**Comment by b-yap** [comment]
except for the failing CI, lgtm.

**Comment by ashneverdawn** [comment]
> except for the failing CI, lgtm.

Oops, my bad, forgot to remove those. Fixed now.

File: runtime/foucoco/src/lib.rs [hunk 1]
```diff
@@ -1293,12 +1293,10 @@ where
...
-				let blockchain = price_feed_request.0;
-				let symbol = price_feed_request.1;
-				let price_feed = <dia_oracle::Pallet<T> as DiaOracle>::get_coin_info(
-					blockchain.to_vec(),
-					symbol.to_vec(),
-				);
+				let blockchain = trim_trailing_zeros(&price_feed_request.0).to_vec();
+				let symbol = trim_trailing_zeros(&price_feed_request.1).to_vec();
+
+				let price_feed = <dia_oracle::Pallet<T> as DiaOracle>::get_coin_info(blockchain, symbol);
```
[the remark this answers is not in the document]
Comment (ashneverdawn) [reply runtime/foucoco/src/lib.rs:1299]: Thanks!

**Comment by ashneverdawn** [comment]
Closing this PR for now, will create a new one with everything once all issues are resolved.

**Pull request closed without merging**